src/main/java/org/candlepin/subscriptions/metering/BaseMeteringResource.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.metering;
+
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.Optional;
+import org.candlepin.subscriptions.util.ApplicationClock;
+import org.springframework.util.StringUtils;
+
+/** Base class for metering resources. */
+public class BaseMeteringResource {
+
+  protected ApplicationClock clock;
+
+  public BaseMeteringResource(ApplicationClock clock) {
+    this.clock = clock;
+  }
+
+  protected OffsetDateTime getDate(String dateToParse) {
```

Comment:
Can you put these methods into a utility class?  I think that will be a cleaner architecture than having the Metering classes inherit from a base class that really just has utility methods in it.

Reply:
Done as we discussed.